docs: fix sync description (#5605)
juicedata/juicefs: docs/en/guide/sync.md

juicefs sync --match-full-path --include='**.gz' --exclude='*' s3://xxx jfs://VOL_NAME/

# Copy all files that do not end with .gz
juicefs sync --match-full-path --exclude='**.gz' s3://xxx/ jfs://VOL_NAME/

# Copy all files except the subdirectory named tempdir
juicefs sync --match-full-path --include='*' s3://xxx/ jfs://VOL_NAME/
```

## Pattern matching {#pattern-matching}

> You can use `--exclude` and `--include` for filtering. If no filtering rules are provided, all files are scanned and copied (`--include='*'` is the default). However, if you were to use `--include` filter to only handle files with the specified pattern, then you must also use `--exclude` as well, see examples above for reference.

Reviewer: ```suggestion
You can use `--exclude` and `--include` for filtering. If no filtering rules are provided, all files are scanned and copied (`--include='*'` is the default). However, if you use the `--include` filter to match files with a specific pattern, you must also use `--exclude` to exclude other files. See the examples above for reference.
```
